extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcConfig.java

```diff
@@ -177,6 +177,22 @@ public class ArcConfig {
     @ConfigItem(defaultValue = "true")
     public boolean detectWrongAnnotations;
 
+    /**
+     * If set to {@code true}, the container will perform additional validations mandated by the CDI specification.
+     * Some improvements on top of the CDI specification may be disabled. Applications that work as expected
+     * in the strict mode should work without a change in the default, non-strict mode.
+     * <p>
+     * The strict mode is mainly introduced to allow passing the CDI Lite TCK. Applications are recommended
+     * to use the default, non-strict mode, which makes CDI more convenient to use. The "strictness" of
+     * the strict mode (the set of additional validations and the set of disabled improvements on top of
+     * the CDI specification) may change over time.
+     * <p>
+     * Note that {@link #transformUnproxyableClasses} also has effect on specification compatibility.
```

Comment:
I originally thought that enabling strictness would automatically disable unproxyable classes transformation, but I didn't really feel like doing it in the end. I don't really have a reason, it just felt strange, but I'd be happy to reconsider.

Reply:
Good point, I'll mention unused beans removal in the javadoc too.

Also, I thought perhaps https://quarkus.io/guides/cdi-reference should list the additional validations performed in strict mode (this PR adds 3), as well as the set of improvements disabled in strict mode (this PR has none). WDYT?

Reply:
I think that it would make sense to mention the strict mode (and its "value") in the docs. I'm not so sure whether it's necessary to list all the validations though...

Reply:
Fair enough. I'll add a paragraph on the strict mode, but won't go into details.

Reply:
Done.